src/try_filter.rs

```diff
@@ -0,0 +1,75 @@
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+
+/// Extension for `Iterator<Item = Result<O, E>>` to filter the Ok(_) and leaving the Err(_) as
+/// is, but allowing the filter to return a `Result<bool, E>` itself
+pub trait TryFilter<O, E>: Sized {
+    /// Filters every `Ok`-value with a function that can return an Err.
+    /// Useful when the filter condition uses functions that can fail.
+    ///
+    ///```
+    ///use resiter::try_filter::TryFilter;
+    ///    use std::str::FromStr;
+    ///
+    ///    let v = ["1", "2", "4", "a", "5"]
+    ///        .iter()
+    ///        .map(Ok)
+    ///        .try_filter_ok(|e| usize::from_str(e).map(|txt| txt < 3))
+    ///        .collect::<Vec<Result<_, _>>>();
+    ///
+    ///    assert_eq!(v.len(), 3);
+    ///    assert_eq!(v.iter().filter(|x| x.is_ok()).count(), 2);
+    ///    assert_eq!(v.iter().filter(|x| x.is_err()).count(), 1);
```

Comment:
```suggestion
    /// use resiter::try_filter::TryFilter;
    /// use std::str::FromStr;
    ///
    /// let v = ["1", "2", "4", "a", "5"]
    ///     .iter()
    ///     .map(Ok)
    ///     .try_filter_ok(|e| usize::from_str(e).map(|txt| txt < 3))
    ///     .collect::<Vec<Result<_, _>>>();
    ///
    /// assert_eq!(v.len(), 3);
    /// assert_eq!(v.iter().filter(|x| x.is_ok()).count(), 2);
    /// assert_eq!(v.iter().filter(|x| x.is_err()).count(), 1);
```

Please fix this here and squash the fix into fe92ac7.

Reply:
Thank you for being thorough. Makes me fee safer that you wouldn't let crazy mistakes of mine through :)